src/App.tsx

```diff
@@ -1,61 +1,138 @@
+import React, { useState } from 'react';
 import './App.scss';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+import { Todo } from './types/todo';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+function getUserById(userById: number) {
+  return usersFromServer.find(user => user.id === userById)
+      || null;
+}
+
+const initialTodos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUserById(todo.userId),
+}));
+
+function getNewTodoId(todos: Todo[]) {
+  const maxId = Math.max(...todos.map(todo => todo.id));
+
+  return maxId + 1;
+}
 
 export const App = () => {
+  const [title, setTitle] = useState('');
+  const [hasTitleError, setHasTitleError] = useState(false);
+  const [userId, setUserId] = useState(0);
+  const [hasUserId, setHasUserId] = useState(false);
+  const [todos, setTodos] = useState<Todo[]>(initialTodos);
+
+  const addTodo = ({ id, ...data }: Todo) => {
+    const newTodo = {
+      id: getNewTodoId(todos),
+      ...data,
+    };
+
+    setTodos(currentTodo => [...currentTodo, newTodo]);
+  };
+
+  const handleTitleChamge = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value);
+    setHasTitleError(false);
+  };
+
+  const handleUserIdChamge = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setUserId(+event.target.value);
+    setHasUserId(false);
+  };
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    setHasTitleError(!title);
+    setHasUserId(!userId);
+
+    if (!title || !userId) {
+      return;
+    }
+
+    const newTodo = {
+      id: Math.random(),
```

Comment:
Are you sure you need to pass it here if you override it in addTodo? Try to simplify this logic.